usr/lib/api/api_interface.c

```diff
@@ -56,22 +56,37 @@ void api_init();
 
 API_Proc_Struct_t *Anchor = NULL;       // Initialized to NULL
 unsigned int Initialized = 0;   // Initialized flag
-pthread_mutex_t GlobMutex;      // Global Mutex
+pthread_mutex_t GlobMutex = PTHREAD_MUTEX_INITIALIZER; // Global Mutex
 CK_FUNCTION_LIST FuncList;
 
 int slot_loaded[NUMBER_SLOTS_MANAGED];  // Array of flags to indicate
                                        // if the STDLL loaded
 
-// For linux only at this time... if it works out we can get rid
-// of the stupid pid tracking.... Linux we kind of have to do this
-// since new threads are processes also, and we will be hosed
+CK_BBOOL in_child_fork_initializer = FALSE;
+
 void child_fork_initializer()
 {
-    if (Anchor) {
-        free(Anchor);
-        Anchor = NULL;
-    }
-}
+    /*
+     * Reinitialize trace so that the trace output appears under the new
+     * process's trace file, and not in the parent process's once.
+     * C_Finalize will pass the new trace handle to the tokens, so that their
+     * finalization code will also trace into the new trace file.
+     */
+    trace_finalize();
+    trace_initialize();
+    /*
+     * Terminate all slots. This will also free the Anchor.
```

Comment:
Anchor is also set to NULL after free ? If so id probably state that explicitely here as well, since Anchor is checked for NULL.

Reply:
added this to the comment.